src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeEditorService.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@RequiredArgsConstructor
+public class SchemeEditorService {
+    private final RestTemplate restTemplate;
+    private final SchemeService schemeService;
+    private final AwsEncryptionServiceImpl awsEncryptionService;
+    private final SchemeRepository schemeRepo;
+    private final UserServiceConfig userServiceConfig;
+
+    @Value("${user-service.domain}")
+    private String userServiceUrl;
+
+    public Boolean doesAdminOwnScheme(Integer schemeId, Integer adminId) {
+        SchemeEntity scheme = schemeService.findSchemeById(schemeId);
+        List<SchemeEntity> adminSchemes = this.schemeRepo.findByCreatedBy(adminId);
+        if (adminSchemes.contains(scheme)) {
+            return Boolean.TRUE;
+        }
+        return Boolean.FALSE;
+    }
+
+    public List<SchemeEditorsDTO> getEditorsFromSchemeId(Integer schemeId, String authHeader) {
+        SchemeEntity scheme = schemeService.findSchemeById(schemeId);
+        List<GrantAdmin> editors = scheme.getGrantAdmins();
+        Integer createdBy = scheme.getCreatedBy();
+
+        return this.mapEditorListToDto(editors, createdBy, authHeader);
+    }
+
+    private List<SchemeEditorsDTO> mapEditorListToDto(List<GrantAdmin> editors, Integer createdBy, String authHeader){
+        List<String> userSubs = editors.stream()
+                .map(editor -> editor.getGapUser().getUserSub())
+                .toList();
+        List<String> emails = getEmailsFromUserSubBatch(userSubs, authHeader);
+
+        return editors.stream()
+                .map(editor -> {
+                    String email = emails.get(userSubs.indexOf(editor.getGapUser().getUserSub()));
+                    Integer id = editor.getId();
+                    SchemeEditorRoleEnum role = id.equals(createdBy)
+                            ? SchemeEditorRoleEnum.Owner : SchemeEditorRoleEnum.Editor;
+
+                    return SchemeEditorsDTO.builder().id(id).email(email).role(role).build();
+                })
+                .sorted(Comparator.comparing(SchemeEditorsDTO::getRole, Comparator.reverseOrder()))
+                .toList();
+    }
+
+
+    private List<String> getEmailsFromUserSubBatch(final List<String> userSubs, final String authHeader) {
+        final String url = userServiceUrl + "/user-emails-from-subs";
+
+        HttpHeaders requestHeaders = new HttpHeaders();
+
+        requestHeaders.add("Cookie", userServiceConfig.getCookieName() + "=" + authHeader);
+
+        UserEmailRequestDto requestBody = UserEmailRequestDto.builder().userSubs(userSubs).build();
+
+        HttpEntity<UserEmailRequestDto> httpEntity = new HttpEntity<>(requestBody, requestHeaders);
+
+        ParameterizedTypeReference<List<UserEmailResponseDto>> responseType =
+                new ParameterizedTypeReference<>() {};
+
+        ResponseEntity<List<UserEmailResponseDto>> response = restTemplate.exchange(
```

Comment:
Did rest template cause us headaches previously (The keep-alive connection)? Might want to check in to see if we still want to use it

Reply:
Is keep-alive enabled by default in spring? AFAIK This was only an issue when connecting to contentful because it provided the header with no expiry as a default

Reply:
We should use webClientBuilder for making requests now rather than restTemplate.

Found this article that summarises the difference pretty well: https://www.baeldung.com/spring-webclient-resttemplate

There should be loads of examples in user-service of web client

Reply:
Sorry, Just realised Dom left the same comment below